.github/workflows/binary_artifact.yml

```diff
@@ -1,71 +1,67 @@
 name: Build binary artifacts
 
 on:
+  pull_request:
+    paths:
+      - '**/binary_artifact.yml'
   push:
     tags:
       - weekly.**
       - 0.**
 
-jobs:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref_name != 'master' }}
 
+jobs:
   build-linux:
     runs-on: ubuntu-20.04
     env:
       CC: gcc
       ZIPNAME: v_linux.zip
     steps:
       - uses: actions/checkout@v4
-      - name: Compile
+      - name: Compile release binaries
         run: |
           make
           ./v -skip-unused -cc $CC -prod -o v cmd/v
           ./v -skip-unused -cc $CC -prod cmd/tools/vup.v
           ./v -skip-unused -cc $CC -prod cmd/tools/vdoctor.v
       - name: Remove excluded
-        run: |
-           rm -rf .git/
-           rm -rf thirdparty/tcc/.git/
-           rm -rf vc/
-           rm -rf v_old
-           rm -rf vlib/v/tests/bench/gcboehm/*.svg
+        run: rm -rf .git/ thirdparty/tcc/.git vc/ v_old vlib/v/tests/bench/gcboehm/*.svg
       - name: Create ZIP archive
         run: |
-           cd ..
-           zip -r9 --symlinks $ZIPNAME v/
-           mv $ZIPNAME v/
-           cd v/
+          cd ..
+          zip -r9 --symlinks $ZIPNAME v/
+          mv $ZIPNAME v/
+          cd v/
       - name: Create artifact
         uses: actions/upload-artifact@v4
         with:
           name: linux
           path: ${{ env.ZIPNAME }}
 
   build-macos-x86_64:
-    runs-on: macos-latest
+    runs-on: macos-13
     env:
       CC: clang
       ZIPNAME: v_macos_x86_64.zip
     steps:
       - uses: actions/checkout@v4
-      - name: Compile
+      - name: Compile release binaries
         run: |
           make
           ./v -skip-unused -cc $CC -prod -o v cmd/v
           ./v -skip-unused -cc $CC -prod cmd/tools/vup.v
           ./v -skip-unused -cc $CC -prod cmd/tools/vdoctor.v
       - name: Remove excluded
-        run: |
-           rm -rf .git/
-           rm -rf thirdparty/tcc/.git/
-           rm -rf vc/
-           rm -rf v_old
-           rm -rf vlib/v/tests/bench/gcboehm/*.svg
+        run: rm -rf .git/ thirdparty/tcc/.git vc/ v_old vlib/v/tests/bench/gcboehm/*.svg
```

Comment:
Restore the line by line script. It is easier to maintain, and has smaller diffs.